corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js

```diff
@@ -107,8 +107,9 @@ hqDefine("cloudcare/js/formplayer/menus/controller", function () {
     var showMenu = function (menuResponse) {
         var menuListView = menusUtils.getMenuView(menuResponse);
         var appPreview = FormplayerFrontend.currentUser.displayOptions.singleAppMode;
-        var sidebarEnabled = toggles.toggleEnabled('SPLIT_SCREEN_CASE_SEARCH') && !appPreview;
-
+        var queryResponse = menuResponse.queryResponse;
+        const hasSearchInputs = menuResponse.type === "query" ? menuResponse.length > 0 : (menuResponse.queryResponse != null && menuResponse.queryResponse.displays.length > 0);
```

Comment:
This conditional check seems very similar to the one you added in `getMenuView`. Is there a way to consolidate it and do the check in one place?

Reply:
Possibly. It does look a bit redundant at first glance. I've done a little digging, but need to do more research to see if we could eliminate some of the repetition. @stephherbers you might have a better idea of how these check interact?

Reply:
hm it did start off with a very different purpose but now has gotten jumbled-- maybe the best approach for this PR would be separating out the `sidebarEnabled` enabled logic to a separate function and we can readdress the further refactoring in a different PR

Reply:
added util method here https://github.com/dimagi/commcare-hq/pull/33876/commits/ae5170f55c00c42051f55c792d722236cb477c0f